src/rich-text-editor.component.js

```diff
@@ -59,6 +59,23 @@ export const RichTextEditor = forwardRef((props, editorRef) => {
     }
   }, [props.unchangedInterval, props.save, divRef.current, focused])
 
+  useEffect(() => {
+    // Clicking on bandicoot richtext buttons triggers a blur event that will setFocus to false we want to delay the
+    // save event that is triggered by blur events. 100ms is arbitrary. Whenever react rerenders the rich-text-editor
+    // due to focused state changing we need to either clear the blurTimeout to prevent a save action from firing 
+    // (in the case of a quick refocus triggered by the rich text buttons) or fire a save event after waiting 100ms
+    if (focused === false) {
+      clearTimeout(blurTimeout.current)
+      blurTimeout.current = setTimeout(() => {
```

Comment:
I think this could be switched to just a variable instead of a ref, and then depend on javascript closures to make sure we always clear the correct timeout:

```js
useEffect(() => {
  const timeoutId = setTimeout(() => {}, 100)
  return () => clearTimeout(timeoutId)
})
```